pandas/core/generic.py

```diff
@@ -230,6 +230,15 @@ def _init_mgr(cls, mgr, axes=None, dtype=None, copy=False):
                 mgr = mgr.astype(dtype=dtype)
         return mgr
 
+    @classmethod
+    def _from_mgr(cls, mgr: BlockManager):
```

Comment:
can you add the return type

Reply:
Would it be NDFrame or FrameOrSeries?

Reply:
I _think_
```suggestion
    def _from_mgr(cls: Type[FrameOrSeries], mgr: BlockManager) -> FrameOrSeries:
```

though not checked

Reply:
looks like we need to change from classmethod in order to get `_constructor`, unless we want to do something like

```
constructor = cls._constructor.fget(cls)
```

which seems sketchy

Reply:
does obj._constructor -> obj._constructor._from_mgr not work with the classmethod. obj._constructor returns a class type not an instance?